Add devcontainer for VS Code - Remote Containers

catunlock opened the pull request:
This adds a docker-compose with four containers, one for the Sardana development with a clean conda environment specified in the file environment.yml and the other 3 related to have a Tango CS (tangodb, databaseds and tangotest).

By default the TANGO_HOST is set to databaseds:10000 but an external Tango DB could be used.

13bscsaamjad commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,43 @@
+	// "postCreateCommand": "apt-get update && apt-get install -y curl",
+
+	// Uncomment to connect as a non-root user if you've added one. See https://aka.ms/vscode-remote/containers/non-root.
+	"remoteUser": "vscode"
```
i recommend using `root` here; its a dev environment and it can require to install various things, this user does not have write access to conda env, hence requiring to build the image again which takes some time!

13bscsaamjad commented on .devcontainer/sardana_dev/environment.yml:
```diff
@@ -0,0 +1,13 @@
+  - pytango=9.3.3
+  - boost
+  - boost-cpp
+  - pip=21.1
```
`itango`/`ipython` is required for spock!

13bscsaamjad replied:
sardana utils like `expconf` require x-server to work, its tricky to make that work in containers; but we can make that part work as well; its really helpful for me atleast to use those during the dev process.

13bscsaamjad reviewed (commented):
i recommend using [mamba](https://github.com/mamba-org/mamba); it is much faster than `conda`!

reszelaz commented on .devcontainer/sardana_dev/environment.yml:
```diff
@@ -0,0 +1,13 @@
+  - boost-cpp
+  - pip=21.1
+  - pip:
+    - pydevd=2.4.1
```
I'm not an expert in pydevd, but accoring tot the [pydevd README](https://github.com/fabioz/PyDev.Debugger/) it should not be necessary to insall it:

> In general, the debugger backend should NOT be installed separately if you're using an IDE which already bundles it (such as PyDev, PyCharm or bundled through debugpy, which is the debug adapter used in VSCode Python and Visual Studio Python).

catunlock commented on .devcontainer/sardana_dev/Dockerfile:
```diff
@@ -0,0 +1,25 @@
+
+FROM mcr.microsoft.com/vscode/devcontainers/miniconda:0-3
+
+# [Option] Install Node.js
```
No. Sorry I was still learning how to use this image of miniconda.

catunlock commented on .devcontainer/sardana_dev/Dockerfile:
```diff
@@ -0,0 +1,25 @@
+#     && pipx reinstall-all
+
+# [Optional] Uncomment this section to install additional OS packages.
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
```
Maybe to develop Sardana only no, but they are useful if you install Pytango from source. But I can remove them if you prefer to have a clean environment and only install them if needed.

catunlock commented:
I added X Forwarding to the container, but you have to run `xhost local:root` in the host.  I also changed the image of the Sardana dev container to micromamba.

reszelaz reviewed (approved):
I have tried it with VSC and it works like a charm!
I will proceed with the merge.
Thanks @catunlock and @13bscsaamjad for your great work!
